addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/internal/protocol/serial/MySensorsSerialConnection.java

```diff
@@ -0,0 +1,199 @@
+    private int baudRate = 115200;
+    private int sendDelay = 0;
+
+    private NRSerialPort serialConnection = null;
+
+    public MySensorsSerialConnection(MySensorsBridgeHandler bridgeHandler, String serialPort, int baudRate,
+            int sendDelay) {
+        super(bridgeHandler);
+
+        this.serialPort = serialPort;
+        this.baudRate = baudRate;
+        this.sendDelay = sendDelay;
+    }
+
+    /**
+     * Tries to accomplish a connection via a serial port to the serial gateway.
+     */
+    @Override
+    public boolean _connect() {
+        logger.debug("Connecting to {} [baudRate:{}]", serialPort, baudRate);
+
+        boolean ret = false;
+
+        updateSerialProperties(serialPort);
+        // deleteLockFile(serialPort);
+
+        serialConnection = new NRSerialPort(serialPort, baudRate);
+        if (serialConnection.connect()) {
+            logger.debug("Successfully connected to serial port.");
+
+            try {
+                logger.debug("Waiting {} seconds to allow correct reset trigger on serial connection opening",
+                        MySensorsBindingConstants.RESET_TIME / 1000);
+                Thread.sleep(MySensorsBindingConstants.RESET_TIME);
+            } catch (InterruptedException e) {
+                logger.error("Interrupted reset time wait");
+            }
+
+            mysConReader = new MySensorsSerialReader(serialConnection.getInputStream(), this);
+            mysConWriter = new MySensorsSerialWriter(serialConnection.getOutputStream(), this, sendDelay);
+
+            ret = startReaderWriterThread(mysConReader, mysConWriter);
+        } else {
+            logger.error("Can't connect to serial port. Wrong port?");
+        }
+
+        return ret;
+    }
+
+    /**
+     * Initiates a clean disconnect from the serial gateway.
+     */
+    @Override
+    public void _disconnect() {
+        logger.debug("Shutting down serial connection!");
+
+        if (mysConWriter != null) {
+            mysConWriter.stopWriting();
+            mysConWriter = null;
+        }
+
+        if (mysConReader != null) {
+            mysConReader.stopReader();
+            mysConReader = null;
+        }
+
+        if (serialConnection != null) {
+            try {
+                serialConnection.disconnect();
+            } catch (Exception e) {
+            }
+            serialConnection = null;
+        }
+
+    }
+
+    /**
+     * This method delete lock file (if present).
+     * RXTX library not removes them after call NRSerialPort.disconnect().
+     * If lock file was not removed, new connection fail to startup
+     *
+     * @param devName is the device used as COM/UART port
+     */
+    @SuppressWarnings("unused")
+    private void deleteLockFile(String devName) {
+        try {
+            String[] namePart = devName.split("/");
+            File lockFile = new File("/var/lock/LCK.." + namePart[namePart.length - 1]);
+            if (lockFile.exists()) {
```

Comment:
The path /var/lock/... only exists on Unix, right? Does this serial interface library work on Windows or MacOS?